### ISSUE TITLE
Unification of `mudata.read` and `anndata.read`

### ISSUE THREAD

**Issue by MUCDK** [scverse/mudata#76]
First, I think documentation could be improved for such a basic function, second, I think it should take similar arguments to `anndata.read`.

**Comment by gtca** [comment]
Thanks, @MUCDK!

Which other arguments from `anndata.read` would you like to see here? Is it about propagating arguments like `as_sparse`, `as_sparse_fmt`, and `chunk_size` to the readers of individual modalities?

**Comment by selmanozleyen** [comment]
@gtca I'd imagine adding the `backup_url` to be very useful. We currently are planning to use `scanpy.readwrite._check_datafile_present_and_download` to download MuData objects and read them but I think it makes more sense for this project to use the private functions of scanpy to load the MuData object.

**Comment by gtca** [comment]
I don't think we should make MuData depend on scanpy though.
If we reuse `scanpy.readwrite._check_datafile_present_and_download`, would it make sense to have it in `muon.read()` instead?

**Comment by selmanozleyen** [comment]
I don't think it matters which function we use as long as `backup_url` functionality is there and similar to scanpy in `muon.read()`. I didn't know muon didn't depend on scanpy so a re-implementation would make sense I guess. From what I saw it was a quite robust implementation
